subsys/bluetooth/host/l2cap.c

```diff
@@ -872,6 +872,11 @@ static void l2cap_chan_tx_process(struct k_work *work)
 		if (sent < 0) {
 			if (sent == -EAGAIN) {
 				ch->tx_buf = buf;
+			} else {
+				/* The ownership of the buffer is with the stack
+				 * since bt_l2cap_chan_send has returned success
+				 */
+				net_buf_unref(buf);
```

Comment:
Sorry, but this is a bit confusing to me. On two levels.

Firstly, these branches handle the return value of l2cap_chan_le_send_sdu(), so why the reference to bt_l2cap_chan_send() in the comment? Was that a typo or intentional?

Secondly, "since bt_l2cap_chan_send has returned success" looks very odd considering that we're in a branch where l2cap_chan_le_send_sdu() has *not* returned success. I'd have worded it the other way around "If l2cap_chan_le_send_sdu() fails the ownership of the buffer is retained by the caller, so we must unref it" or something like that.

Reply:
Yes, this is intentional. The ownership has been transferred by a previous user call of `bt_l2cap_chan_send` (not the internal `l2cap_chan_le_send_sdu`). `bt_l2cap_chan_send` returned success to the caller, otherwise the buffer would not have been queued.

Reply:
Comment was removed again. So I resolve this :)